### PR TITLE
Bump the compiler version to 4.2 (and 3.4)

### PR DESCRIPTION
**Pull request by jrose-apple**
Whether there's going to be an actual 4.2 release or we go straight to 5.0, we're not 4.1 anymore.

### CODE REVIEW

**Comment by jrose-apple** [comment]
cc @aciidb0mb3r

**Comment by jrose-apple** [comment]
@swift-ci Please test

**Comment by aciidb0mb3r** [comment]
Thanks Jordan!

**Comment by bob-wilson** [comment]
This looks OK to me. Should we just go straight to 5.0 now?

**Comment by jrose-apple** [comment]
Bah, missed a test. Uh, it's probably okay to go straight to 5 but I wonder at what point we're going to discuss the testing strategy. (The tests are mostly still run in Swift 3 mode unless specified otherwise because they were a corpus of Swift 3 code.)

**Comment by jrose-apple** [comment]
There's also "get the compiler to advertise itself as Swift 5" vs. "change the default when you omit -swift-version" and those don't have to be in sync…but it makes more sense when they are.

**Comment by jrose-apple** [comment]
@swift-ci Please test

**Comment by jrose-apple** [comment]
Since #13843 didn't quite go cleanly and I haven't had time to look at it, can we merge this in the mean time?

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test

File: utils/build_swift/defaults.py [hunk 1]
```diff
@@ -40,7 +40,7 @@
 CMAKE_GENERATOR = 'Ninja'
 
 COMPILER_VENDOR = 'none'
-SWIFT_USER_VISIBLE_VERSION = CompilerVersion('4.1')
+SWIFT_USER_VISIBLE_VERSION = CompilerVersion('4.2')
 CLANG_USER_VISIBLE_VERSION = CompilerVersion('5.0.0')
```
Comment (jrose-apple) [inline utils/build_swift/defaults.py]: This default is very suspicious. Shouldn't we be getting it from the Clang repository somehow? cc @Rostepher

**Comment by jrose-apple** [comment]
Per https://swift.org/blog/4-2-release-process/, this is the right one to commit now, right @tkremenek?

@swift-ci Please smoke test

**Comment by tkremenek** [comment]
Yes

**Pull request merged**